packages/atlas/src/components/_inputs/Input/Input.styles.ts

```diff
@@ -0,0 +1,94 @@
+import { css } from '@emotion/react'
+import styled from '@emotion/styled'
+
+import { cVar, sizes } from '@/styles'
+
+import {
+  InputSize,
+  NodeWidthProps,
+  getInputPseudoSelectorStyles,
+  getPadding,
+  horizontalPadding,
+  sharedInputStyles,
+} from '../inputs.utils'
+
+type TextInputProps = {
+  error?: boolean
+  inputSize: InputSize
+} & NodeWidthProps
+
+export const TextInput = styled.input<TextInputProps>`
+  width: 100%;
+  padding: ${sizes(3)} ${sizes(5)};
+
+  ::-webkit-outer-spin-button,
+  ::-webkit-search-cancel-button,
+  ::-webkit-inner-spin-button {
+    appearance: none;
+    margin: 0;
+  }
+
+  &[type='number'],
+  &[type='search'] {
+    appearance: textfield;
+  }
+
+  ${({ inputSize, leftNodeWidth, rightNodeWidth }) => getPadding(inputSize, { leftNodeWidth, rightNodeWidth })};
+
+  ${({ error }) => getInputPseudoSelectorStyles({ error })};
+`
+
+type NodeContainerProps = {
+  left?: boolean
+  disabled?: boolean
+  isButton?: boolean
+  isFocused?: boolean
+  size: InputSize
+}
+const nodePlacementStyles = ({ left, size, isButton }: NodeContainerProps) =>
+  left
+    ? css`
+        left: 0;
+        padding-left: calc(${horizontalPadding[size]});
+      `
+    : css`
+        right: 0;
+        padding-right: calc(${horizontalPadding[size]} - ${isButton ? sizes(1) : '0px'});
```

Comment:
Maybe better to have the condition outside of `calc` so we don't do `calc(100px - 0px)`?